### ISSUE TITLE
Question: Setting one instance of a desktop to tile and another to float

### ISSUE THREAD

**Issue by atidyshirt** [lingtjien/Grid-Tiling-Kwin#45]
Hey, I am wondering if it is possible to set up a desktop so that when windows are added they are floating, and then on another desktop have them so they tile.

Also, I think it would be great if you added a feature to toggle on/off tiling for all windows.

Awesome script

Many thanks,

Jordan

**Comment by lingtjien** [comment]
> Hey, I am wondering if it is possible to set up a desktop so that when windows are added they are floating, and then on another desktop have them so they tile.

Currently not possible, all virtual desktops are treated the same with the exception of virtual desktops on different (physical) screens, which allow you to specify the grid size.

> Also, I think it would be great if you added a feature to toggle on/off tiling for all windows.

With respect to all currently open windows or only newly added ones?

**Comment by atidyshirt** [comment]
I mean with respect to all new windows after toggle is turned on.

**Comment by Hawkeye0021** [comment]
I would also like to see this feature, or a way to make it so all windows float by default, and can then be selectively tiled. As apposed to tiled by default and selectively de-tiled.

**Comment by GitSquared** [comment]
I'm also interested in this. Might open a (third) PR tomorrow.

**Comment by lingtjien** [comment]
The PR is concerning the floating by default on certain virtual desktops.

My problem with all this is that, now the script is already providing floating by default for clients specifically specified, then you also want to add floating by default for specific desktops and maybe in the future for specific screens and specific activities... IDK, this seems like a lot of different ways to specify floating by default behavior and thus requiring a lot of extra code... 

I'm leaning more towards @Hawkeye0021 suggestion of only adding a toggle that makes clients float by default instead of tile (which requires very minimal amount of code) and what I think catches most of the use cases.

**Comment by lingtjien** [comment]
You can try the dev branch, added a global shortcut `Grid-Tiling: Toggle Tile` (Default `Meta+Shift+T`) which should switch between tiling by default and floating by default

**Comment by GitSquared** [comment]
@lingtjien What the PR does UX-wise is provide a on/off switch for the whole plugin, that affects only the current desktop.

Toggling spawning behavior between tile and float seems like a very different feature to me.

**Comment by lingtjien** [comment]
@GitSquared yes and I said that in my initial response, but then please answer this question:

Why do you want to provide a tiling toggle per desktop only, why not per client (that already existed, that I now removed)? Why not per physical screen? Why not per activity?

Personally, I feel like providing all these *specific* different ways to specify clients to float by default instead of tile is a bit too much. Hence I went with @Hawkeye0021 suggestion of providing a global shortcut only so that the user can decide on the fly if they want to tile now or float. However, it's just an idea, I'm open to suggestions.

TLDR: Global tiling/float by default toggle seems more useful to me than specifying it per client/desktop/screen/activity/etc...

**Comment by Hawkeye0021** [comment]
@lingtjien awesome thank you! I'm very excited to start using this script again. Having everything open up tiled messed things up pretty quick especially with things that have minimum sizes like discord and system settings. Floating by default and being able to tile a couple of main windows I'm working with is great!

**Comment by GitSquared** [comment]
@lingtjien Because virtual desktops are the smallest client group you can work with and depending on what I have on each one I may or may not need tiling.

But that's just how I work and I understand you might not like it - I'm using my fork either way.

**Comment by lingtjien** [comment]
Of course, that's the beauty of open source after all. I'm just trying to get a better understanding of your workflow and why it is that you want tiling for specific virtual desktops. Maybe that can improve my workflow and this script as well.

**Comment by lingtjien** [comment]
for now sticking with the global toggle for starting in tile/float mode since I believe that offers the most flexibility
